### PR TITLE
Run tests automatically in the PR checks workflow (PR)

### PR DESCRIPTION
**Pull request by deyanzz**
**Description**:
This PR adds commands to start and stop the local node during the PR checks.
Updated the YAML script

**Related issue(s)**: 

- https://github.com/hashgraph/hedera-sdk-cpp/issues/201
- https://github.com/hashgraph/hedera-sdk-cpp/issues/164

Fixes https://github.com/hashgraph/hedera-sdk-cpp/issues/257

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/tests/LedgerIdTest.cc [hunk 1]
```diff
@@ -128,15 +128,15 @@ TEST_F(LedgerIdTest, FromString)
...
-}
+// TEST_F(LedgerIdTest, FromBadStrings)
+// {
+//   // Given / When / Then
+//   EXPECT_THROW(const LedgerId ledgerId = LedgerId::fromString("mainet"), std::invalid_argument);
+//   EXPECT_THROW(const LedgerId ledgerId = LedgerId::fromString("this is a bad string"), std::invalid_argument);
+//   EXPECT_THROW(const LedgerId ledgerId = LedgerId::fromString("1234567890abcdefg"), std::invalid_argument);
+//   EXPECT_THROW(const LedgerId ledgerId = LedgerId::fromString("TESTNET"), std::invalid_argument);
+//   EXPECT_THROW(const LedgerId ledgerId = LedgerId::fromString("Previewnet"), std::invalid_argument);
+// }
```
Comment (rwalworth) [inline sdk/tests/LedgerIdTest.cc:139]: Same with this?

File: sdk/tests/ECDSAsecp256k1PublicKeyTest.cc [hunk 2]
```diff
@@ -216,24 +216,24 @@ TEST_F(ECDSAsecp256k1PublicKeyTest, CompressBytes)
...
+
+//   // Throw if not correct size
+//   EXPECT_THROW(const std::vector<std::byte> bytes =
+//                  ECDSAsecp256k1PublicKey::uncompressBytes({ std::byte(0x3), std::byte(0x2), std::byte(0x1) }),
+//                std::invalid_argument);
+//   // Throw if not starting with 0x2 or 0x3
+//   EXPECT_THROW(const std::vector<std::byte> bytes = ECDSAsecp256k1PublicKey::uncompressBytes(
+//                  std::vector<std::byte>(ECDSAsecp256k1PublicKey::COMPRESSED_KEY_SIZE, std::byte(0x1))),
+//                std::invalid_argument);
+// }
```
[the remark this answers is not in the document]
Comment (deyanzz) [reply sdk/tests/ECDSAsecp256k1PublicKeyTest.cc:236]: This test failed with the following error:

> Start 133: ECDSAsecp256k1PublicKeyTest.UncompressBytes
133/362 Test #133: ECDSAsecp256k1PublicKeyTest.UncompressBytes ..........................................................................................................................................................................................................................................................................................................................................Subprocess aborted***Exception:   0.01 sec
Running main() from ./googletest/src/gtest_main.cc
Note: Google Test filter = ECDSAsecp256k1PublicKeyTest.UncompressBytes
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from ECDSAsecp256k1PublicKeyTest
[ RUN      ] ECDSAsecp256k1PublicKeyTest.UncompressBytes
[       OK ] ECDSAsecp256k1PublicKeyTest.UncompressBytes (0 ms)
[----------] 1 test from ECDSAsecp256k1PublicKeyTest (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 1 test.

[ FATAL ] ./googletest/include/gtest/internal/gtest-port.h:1774:: pthread_key_delete(key_)failed with error 22

You can see the failed PR check over [here](https://github.com/hashgraph/hedera-sdk-cpp/actions/runs/4575042834/jobs/8077398839).

My suggestion is to leave this test commented and to create a separated issue to track it. We will dig further on it next week.

File: sdk/tests/LedgerIdTest.cc (hunk 1 above)
Comment (deyanzz) [reply sdk/tests/LedgerIdTest.cc:139]: This test failed with the following error:

> Start 219: LedgerIdTest.FromBadStrings
219/361 Test #219: LedgerIdTest.FromBadStrings ..........................................................................................................................................................................................................................................................................................................................................................Subprocess aborted***Exception:   0.01 sec
Running main() from ./googletest/src/gtest_main.cc
Note: Google Test filter = LedgerIdTest.FromBadStrings
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from LedgerIdTest
[ RUN      ] LedgerIdTest.FromBadStrings
[       OK ] LedgerIdTest.FromBadStrings (0 ms)
[----------] 1 test from LedgerIdTest (0 ms total)
[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 1 test.
[ FATAL ] ./googletest/include/gtest/internal/gtest-port.h:1774:: pthread_key_delete(key_)failed with error 22

You can see the failed PR check over [here](https://github.com/hashgraph/hedera-sdk-cpp/actions/runs/4575797534/jobs/8079092766).

My suggestion is the same like above - to leave this test commented and to create a separated issue for fixing it.

**Review by rwalworth — APPROVED** [review]
LGTM

**Comment by deyanzz** [comment]
Create 2 new issues to track the commented tests:

- https://github.com/hashgraph/hedera-sdk-cpp/issues/259
- https://github.com/hashgraph/hedera-sdk-cpp/issues/260

**Pull request merged**